logstash-core/lib/logstash/agent.rb

```diff
@@ -148,10 +148,10 @@ def converge_state_and_update
     @pipelines_mutex.synchronize do
       pipeline_actions = resolve_actions(results.response)
       converge_result = converge_state(pipeline_actions)
+      update_metrics(converge_result)
     end
 
     report_currently_running_pipelines(converge_result)
```

Comment:
interestingly, moving those two lines into the mutex caused more tests to fail, I need to investigate why.